### PR TITLE
feat: mf-6313 swap/bridge in popup

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/mf-6313

### CODE REVIEW

File: packages/mask/popups/pages/Trader/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        [RoutePaths.History]: 'History',
+        [RoutePaths.Confirm]: 'Confirm Swap',
+        [RoutePaths.BridgeConfirm]: 'Confirm Bridge',
+        [RoutePaths.SelectLiquidity]: 'Select Liquidity',
+        [RoutePaths.Slippage]: 'Slippage',
+        [RoutePaths.QuoteRoute]: 'Quote Route',
+        [RoutePaths.BridgeQuoteRoute]: 'Quote Route',
+        [RoutePaths.TradingRoute]: 'Trading Route',
+        [RoutePaths.Exit]: null,
+        [RoutePaths.NetworkFee]: 'Network fee',
```
Comment (swkatmask) [inline packages/mask/popups/pages/Trader/Header.tsx:24]: Skip translation here until we merge it into develop branch, and we will do it with `lingui`.

File: packages/mask/popups/pages/Trader/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import { MaskTabList } from '@masknet/theme'
+import { TabContext } from '@mui/lab'
+import { Tab } from '@mui/material'
+import { useLocation, useNavigate } from 'react-router-dom'
+import { NormalHeader } from '../../components/index.js'
+
+export function Header() {
+    const { mode } = useTrade()
+    const basepath = '/trader'
+    const titleMap: Record<RoutePaths, string | null> = {
```
Comment (guanbinrui) [inline packages/mask/popups/pages/Trader/Header.tsx:13]: i18n?

Comment (UncleBill) [reply packages/mask/popups/pages/Trader/Header.tsx:13]: Explained above

File: packages/mask/popups/components/TokenPicker/index.tsx [hunk 3]
```diff
@@ -20,34 +22,50 @@ const useStyles = makeStyles()((theme) => {
...
+    const { data: okxTokens } = useOKXTokenList(chainId, assetSource === 'okx')
+    const okxAssets = useMemo(() => {
+        if (!okxTokens?.length) return EMPTY_LIST
+        const balanceMap = new Map(standardAssets.map((x) => [x.address.toLowerCase(), x.balance]))
+        // To reduce queries, get balance from standardAssets and patch okxTokens with it
+        return okxTokens.map((x) => {
+            const balance = balanceMap.get(x.address.toLowerCase())
+            return !balance || balance === '0' ? x : { ...x, balance }
+        }) as typeof okxTokens
+    }, [okxTokens, standardAssets])
```
Comment (guanbinrui) [inline packages/mask/popups/components/TokenPicker/index.tsx:102]:
```suggestion
       return okxTokens.map(x => {
           standardAssets.find(y => isSameAddress(x.address, y.address) && y.balance ? { ...x, balance } : x)
       })
```

File: packages/mask/popups/components/TokenPicker/index.tsx [hunk 4]
```diff
@@ -63,35 +81,74 @@ export const TokenPicker = memo(function TokenPicker({
...
+                itemData={{
+                    tokens: availableAssets,
+                    networks: filteredNetworks,
+                    chainId,
+                    address,
+                    onSelect,
+                }}
+                itemKey={(index, data) => {
+                    const asset = data.tokens[index]
+                    return `${asset.chainId}.${asset.address}`
```
Comment (guanbinrui) [inline packages/mask/popups/components/TokenPicker/index.tsx:149]:
```suggestion
                    return `${asset.chainId}.${asset.address.toLowerCase()}`
```

File: packages/mask/popups/components/TokenPicker/index.tsx [hunk 5]
```diff
@@ -63,35 +81,74 @@ export const TokenPicker = memo(function TokenPicker({
...
+    /** Selected address */
+    address?: string
+    onSelect?(asset: Web3Helper.FungibleAssetAll | Web3Helper.FungibleTokenAll): void
+}>
+
+function Row({ data, index, style }: RowProps) {
+    const { tokens, networks, chainId, address, onSelect } = data
+    const asset = tokens[index]
+
+    const network = networks.find((x) => x.chainId === asset.chainId)!
```
Comment (guanbinrui) [inline packages/mask/popups/components/TokenPicker/index.tsx]:
```suggestion
    const network = networks.find((x) => x.chainId === asset.chainId)
    if (!network) return null;
```

File: packages/mask/popups/pages/Trader/useImplementRuntime.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        ): Promise<Web3Helper.FungibleTokenAll | null> => {
+            const supportedChains = chainQuery.data ?? (await chainQuery.refetch()).data
+
+            const picked = await ChooseTokenModal.openAndWaitForClose({
+                // Only from token can decide the chain
+                chainId: ((isSwap ? fromChainId : currentToken?.chainId) || chainId) as ChainId,
+                address: currentToken?.address,
+                chains: supportedChains?.map((x) => x.chainId),
+                lockChainId: isSwap && side === 'to' && !!fromChainId,
+                assetSource: 'okx',
```
Comment (guanbinrui) [inline packages/mask/popups/pages/Trader/useImplementRuntime.ts]:
```suggestion
                assetSource: Source.OKX,
```

File: packages/mask/popups/components/TokenPicker/index.tsx (hunk 4 above)
Comment (swkatmask) [reply packages/mask/popups/components/TokenPicker/index.tsx:149]: `itemKey` is just used as a identifier, besides, `asset.address` is stable, no any runtime modification. I don't think we need to do this.

File: packages/mask/popups/components/TokenPicker/index.tsx (hunk 5 above)
Comment (swkatmask) [reply packages/mask/popups/components/TokenPicker/index.tsx]: `network` is not the critical factor to `TokenItem` but asset is.

File: packages/mask/popups/components/TokenPicker/index.tsx (hunk 3 above)
Comment (swkatmask) [reply packages/mask/popups/components/TokenPicker/index.tsx:102]: Nesting loops, that would be `O(n^2)`, `Map.get` is more efficient.

Comment (swkatmask) [reply packages/mask/popups/components/TokenPicker/index.tsx:102]: There could be thousands of tokens in OKX's token list, so performance would be a thing here.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**